src/delphi/train/tokenizer.py

```diff
@@ -0,0 +1,65 @@
+import os
+
+import sentencepiece as spm
+from datasets import Dataset
+
+
+def get_tokenizer_model_path(
+    vocab_size: int,
+    cache_dir: str = "cache"
+) -> str:
+    """
+    Returns path to the SentencePiece tokenizer model for a given vocab size.
+    """
+    if vocab_size == 0:
+        return ""
+    else:
+        return os.path.join(cache_dir, f"tok{vocab_size}.model")
+
+def train_vocab(
+    vocab_size: int, 
+    dataset: Dataset,
+    cache_dir: str = "cache"
+) -> None:
+    """
+    Trains a custom SentencePiece tokenizer.
+    """
+    assert vocab_size > 0, "Vocab size must be positive"
+    
+    if not os.path.exists(cache_dir):
+        os.makedirs(cache_dir)
+
+    # export text as a single text file
+    text_file = os.path.join(cache_dir, "text.txt")
+    with open(text_file, 'w', encoding='utf-8') as file:
+        for item in dataset:
+            text = item['story']
```

Comment:
feature needs to be configurable. If not set, should default to the only feature there is, or fail if there are more than two

Reply:
added as a parameter